GetJobber/atlantis
# fix(components): Create a hook to focus back on modal and popover activator

## Motivations

Focus back to the element that opened the modal/popover when they close to Improve the accessibility of it for keyboard users.

## Changes



### Added

- A hook called `useRefocusOnActivator` that puts the focus back on the thing that opened the modal/popover when they gets closed
- Added documentation and test for said hook

### Changed

- Implemented `useRefocusOnActivator` hook on Modal and Popover

### Deprecated

- 

### Removed

- 

### Fixed

- 

### Security

- 

## Testing

- Open a modal or popover
- Close modal or popover via keyboard tabbing/shortcuts or clicking close
- It should bring back the focus to the one that opened it

## Limitations
- Only applies to element that can be focused
- The activator has to open the modal/popover in order for it to be able to focus back
  - This won't work if you programmatically open a dialog by any other means as that would mean it wasn't opened by a user intent.

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

![Random photo of Atlantis](https://loremflickr.com/672/400/atlantis)
